gantry/core/can_task.cpp

```diff
@@ -47,23 +52,37 @@ static auto motion_group_executor_dispatch_target =
         can_move_group_executor_handler};
 
 /** Dispatcher to the various handlers */
-static auto dispatcher = Dispatcher(
+static auto dispatcher = can_task::GantryDispatcherType(
     [](auto _) -> bool { return true; }, motor_dispatch_target,
     motion_group_dispatch_target, motion_group_executor_dispatch_target,
     device_info_dispatch_target);
 
-auto poller =
-    freertos_can_dispatch::FreeRTOSCanReader<1024, decltype(dispatcher)>(
-        can_bus_1, dispatcher);
+auto static reader_message_buffer =
+    freertos_can_dispatch::FreeRTOSCanBufferControl<
+        can_task::reader_message_buffer_size, decltype(dispatcher)>{dispatcher};
 
-[[noreturn]] void task_entry() {
-    interfaces::initialize();
+auto static reader_task_control =
+    freertos_task::FreeRTOSTaskControl<can_task::reader_task_stack_depth>{};
+auto static writer_task_control =
+    freertos_task::FreeRTOSTaskControl<can_task::writer_task_stack_depth>{};
 
-    can_bus_1.setup_node_id_filter(my_node_id);
+auto can_task::start_reader(can_bus::CanBus& canbus)
+    -> can_task::CanMessageReaderTask {
+    LOG("Starting the CAN reader task\n");
 
-    interfaces::get_motor().driver.setup();
+    canbus.setup_node_id_filter(my_node_id);
 
-    poller();
+    auto poller = can_task::CanReaderTaskEntry{canbus, reader_message_buffer};
+    return freertos_task::FreeRTOSTask<can_task::reader_task_stack_depth,
+                                       decltype(poller)>{
+        poller, reader_task_control, 5, "can task"};
 }
 
-auto static task = freertos_task::FreeRTOSTask<512, 5>("can task", task_entry);
+auto can_task::start_writer(can_bus::CanBus& canbus)
```

Comment:
I'm sorry, but I don't understand. Which message writer code do you mean?

Reply:
Sorry I should've been more clear. Since we have separate reader and writer tasks now, I was thinking it might make things a little more readable if we keep individual tasks in separate files. So can task writer and implementation could be in a file called `can_writer_task.cpp` and the reader task could be called `can_reader_task.cpp` etc.

Reply:
Oh. Doesn't matter to me either way. If you prefer two separate files I'll do it.